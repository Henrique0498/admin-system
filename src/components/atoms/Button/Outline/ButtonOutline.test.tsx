import { render, screen } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import theme from '../../../../styles/theme'

import { ButtonOutline } from './index'

describe('<ButtonOutline/>', () => {
  it('should render the heading', () => {
    render(
      <ThemeProvider theme={theme}>
        <ButtonOutline>Button</ButtonOutline>
      </ThemeProvider>
    )

    expect(screen.getByRole('button', { name: /button/i })).toBeInTheDocument()
  })
  it('should render correctly', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <ButtonOutline>Button</ButtonOutline>
      </ThemeProvider>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
