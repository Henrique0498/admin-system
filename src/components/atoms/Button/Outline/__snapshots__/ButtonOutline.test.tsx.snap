// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonOutline/> should render correctly 1`] = `
.c0 {
  padding: 0.625rem 1.25rem;
  font-size: 0.875rem;
  color: #a855f7;
  border: 2px solid #a855f7;
  background: linear-gradient( 90deg, transparent 50%, #a855f7 50% );
  border-radius: 4px;
  font-weight: 400;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-size: 210%;
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

.c0:focus,
.c0:hover {
  background-position-x: 98%;
  color: #fff;
}

.c0:focus {
  box-shadow: 0 0 0 2px rgba(139,92,246,0.5);
}

<button
  class="c0"
  color="purple"
  type="button"
>
  Button
</button>
`
